models/classes/Csv/Factory/ReaderFactory.php

```diff
@@ -0,0 +1,58 @@
+<?php
+
+/**
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2022 (original work) Open Assessment Technologies SA.
+ */
+
+declare(strict_types=1);
+
+namespace oat\tao\model\Csv\Factory;
+
+use League\Csv\Exception;
+use InvalidArgumentException;
+use League\Csv\InvalidArgument;
+use oat\tao\model\Csv\Service\Reader;
+use League\Csv\Reader as LeagueReader;
+
+class ReaderFactory
+{
+    public const DELIMITER = 'delimiter';
+
+    /**
+     * @param resource $stream
+     */
+    public function createFromStream($stream, array $options = []): Reader
+    {
+        if (!is_resource($stream)) {
+            throw new InvalidArgumentException('The provided value must be a resource.');
+        }
+
+        $reader = LeagueReader::createFromStream($stream);
+
+        try {
+            $reader->setHeaderOffset(0);
+
+            if (!empty($options[self::DELIMITER]) && is_string($options[self::DELIMITER])) {
+                $reader->setDelimiter($options[self::DELIMITER]);
+            }
+        } catch (Exception | InvalidArgument $exception) {
```

Comment:
Isn't the same in this case? 

```suggestion
        } catch (Throwable $exception) {
```